src/main/java/patternverifiers/SingletonVerifier.java

```diff
@@ -1,17 +1,339 @@
 package patternverifiers;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
 import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.Modifier;
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.body.ConstructorDeclaration;
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.ast.body.MethodDeclaration;
+import com.github.javaparser.ast.expr.BinaryExpr;
+import com.github.javaparser.ast.expr.MethodCallExpr;
+import com.github.javaparser.ast.expr.ObjectCreationExpr;
+import com.github.javaparser.ast.stmt.IfStmt;
+
+import base.VariableReader;
 
 /**
  * A verifier for the singleton pattern.
  */
 public class SingletonVerifier implements IPatternVerifier {
 
+    private final transient List<ConstructorDeclaration> constDeclList = new ArrayList<>();
+    private transient FieldDeclaration instanceVar;
+    private transient boolean isInstantiated;       // Tbh idk what transient means but PMD would
+    // not let it work without it.
+
     public SingletonVerifier() {
     }
 
     @Override
     public Feedback verify(CompilationUnit compUnit) {
-        throw new UnsupportedOperationException();
+        boolean isValid = callsConstructor(compUnit) && hasAStaticInstance(compUnit) &&
+                          onlyInstantiatedIfNull(compUnit) && hasPrivateConstructor(compUnit) &&
+                          isInstantiated;
+
+        return new Feedback(isValid);
+    }
+
+    /**
+     * Method for checking if all the constructors in a Java class are private.
+     *
+     * @param compUnit The CompilationUnit representing the Java class to look at
+     *
+     * @return True iff all constructors are private
+     */
+    public boolean hasPrivateConstructor(CompilationUnit compUnit) {
+        List<Boolean> listOfIsPrivate = new ArrayList<>();
+        boolean isPrivConstr = true;
+
+        // Finds and checks if the constructors are private or not.
+        compUnit.findAll(ConstructorDeclaration.class).forEach(constructor -> {
+            if (constructor.isPrivate()) {
+                listOfIsPrivate.add(true);
+            } else {
+                listOfIsPrivate.add(false);
```

Comment:
This if statement can be replaced with 
```suggestion
                listOfIsPrivate.add(constructor.isPrivate());
```